web/libs/ui/src/lib/checkbox/checkbox.tsx

```diff
@@ -0,0 +1,59 @@
+/* eslint-disable-next-line */
+import React, { type InputHTMLAttributes } from "react";
+import styles from "./checkbox.module.scss";
+import { nanoid } from "nanoid";
+
+export interface CheckboxProps extends Omit<InputHTMLAttributes<HTMLInputElement>, "type"> {
+  indeterminate?: boolean;
+}
+const classNameMap = { ...styles };
+
+export const Checkbox = ({ checked, indeterminate, style, onChange, children, ...props }: CheckboxProps) => {
+  const checkboxRef = React.createRef();
+  const withLabel = !!children;
+  const [id] = React.useState(nanoid);
```

Comment:
Maybe something more semantic?
```suggestion
  const id = React.useMemo(nanoid, []);
```